src/_common/Catalog/CatalogUtility.cs

```diff
@@ -10,6 +10,77 @@ namespace Skender.Stock.Indicators;
 /// </summary>
 public static class CatalogUtility
 {
+    // Internal type mapping for performance - avoids reflection overhead
+    private static readonly Dictionary<string, Type> TypeMapping = new()
+    {
+        { "AdlResult", typeof(AdlResult) },
```

Comment:
Will this work as a more strongly typed equipment?

```suggestion
        { nameof(AdlResult), typeof(AdlResult) },
```